sdk/webpubsub/Azure.Messaging.WebPubSub/src/HealthApiClient_extensions.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+#pragma warning disable AZC0007
+
+namespace Azure.Messaging.WebPubSub
+{
+    /// <summary> The HealthApi service client. </summary>
+    internal partial class HealthApiClient
+    {
+        /// <summary> Initializes a new instance of HealthApiClient. </summary>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        public HealthApiClient(AzureKeyCredential credential, Uri endpoint = null, WebPubSubServiceClientOptions options = null)
+        {
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
```

Comment:
Nit - use `Argument.NotNull(credential, nameof(credential));` to match the rest of our code base.  You might need an `<Compile Include="$(AzureCoreSharedSources)Argument.cs" LinkBase="Shared" />` in your .csproj.